### PR TITLE
You can no longer turn into a spinning flail by spamming the move up verb

### LINKED ISSUE #49667: You can move upwards as a human onto catwalks
Move upwards verb ascends you magically to the catwalk above you (if there is one) on multi-z maps. This does not seem like intended behavior.

### LINKED ISSUE #49761: You can attempt to climb up and down while dead
## Round ID

[Round ID] 131612

## Testmerges:

#49065: 'Disable library machine uis' by optimumtact at commit f261fc0b45
#49476: 'Adds tackling (VIDEO INSIDE)' by Ryll-Ryll at commit 32fad5b299

## Reproduction:

Attempt to climb up a wall while dead. Fall down. Spam this. Become the meatgrinder. Cry.

(Please don't shoot me bagilites, you'll have at least a couple hours with this.)

### LINKED ISSUE #51006: "Move Upwards" verb is really bad.
## Reproduction:

Play on a multi-z map, move to a spot with open space above you, click the IC tab, and click "move upwards".
You will magically levitate upwards and instantaneously fall back down.
You can use this verb repeatedly very fast, you can use it while stunned, and you can even use it while dead.

### PR DESCRIPTION
**Pull request by ghost**
:cl: qustinnus
fix: you can no longer move up when you are going to fall down anyways, or you know, dead.
/:cl:

why even bother checking whether someone should be able to move up am i right

Fixes #51006
Fixes #49761
Fixes #49667

### CODE REVIEW

**Comment by uomo91** [comment]
Relevant issues:

https://github.com/tgstation/tgstation/issues/51006
https://github.com/tgstation/tgstation/issues/49761
https://github.com/tgstation/tgstation/issues/49667

**Comment by uomo91** [comment]
Will this solve moving downwards while dead/incapacitated, or being able to move upwards to catwalks without a ladder?

**Comment by Dennok** [comment]
Great!
Add this please to Move Down too. Maybe someone create antigravity fields in future.

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -456,6 +456,13 @@
 	set name = "Move Upwards"
 	set category = "IC"
 
+	var/turf/current_turf = get_turf(src)
+	var/turf/above_turf = SSmapping.get_turf_above(current_turf)
+
+	if(can_zFall(above_turf, 1, current_turf, DOWN)) //Will be fall down if we go up?
+		to_chat(src, "<span class='notice'>You are not superman.<span>")
```
Comment (Dennok) [inline code/modules/mob/mob_movement.dm:463]: Maybe something like "You jump on place." is better?
and show it to all near.

Comment (Qustinnus) [reply code/modules/mob/mob_movement.dm:463]: i'd rather not, that's not very universal to all cases were this wouldnt work

**Comment by Qustinnus** [comment]
@Dennok this affects move down too because zmove is used by both

**Comment by Qustinnus** [comment]
@uomo91 yes to first no to second

File: code/modules/mob/mob_movement.dm (hunk 1 above)
Comment (fludd12) [reply code/modules/mob/mob_movement.dm:463]: Does / should this account for creatures that can fly? It would be interesting to see.

Comment (Qustinnus) [reply code/modules/mob/mob_movement.dm:463]: it does

Comment (Qustinnus) [reply code/modules/mob/mob_movement.dm:463]: anything that would float above open space would be able to move up. The only cave-at is that you can move up even if there's nothing to move up fromwhen theres no gravity, but that is something we can work in later.

Comment (fludd12) [reply code/modules/mob/mob_movement.dm:463]: procedural assignment of z-levels to allow for arbitrary heights of locations? 😳

Comment (Qustinnus) [reply code/modules/mob/mob_movement.dm:463]: huh?

Comment (Dennok) [reply code/modules/mob/mob_movement.dm:463]: I say about this check.

Comment (ATH1909) [reply code/modules/mob/mob_movement.dm:463]: shouldn't "superman" be capitalized? it's a proper noun

Comment (ShizCalev) [inline code/modules/mob/mob_movement.dm:463]: capitalize it please

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob_movement.dm (hunk 1 above)
Comment (Qustinnus) [reply code/modules/mob/mob_movement.dm:463]: you mean superman?

**Pull request merged**